starlette/staticfiles.py

```diff
@@ -158,20 +157,24 @@ async def get_response(self, path: str, scope: Scope) -> Response:
         raise HTTPException(status_code=404)
 
     def lookup_path(
-        self, path: str
-    ) -> typing.Tuple[str, typing.Optional[os.stat_result]]:
+        self, path: Path
+    ) -> typing.Tuple[Path, typing.Optional[os.stat_result]]:
         for directory in self.all_directories:
-            full_path = os.path.realpath(os.path.join(directory, path))
-            directory = os.path.realpath(directory)
-            if os.path.commonprefix([full_path, directory]) != directory:
-                # Don't allow misbehaving clients to break out of the static files
-                # directory.
-                continue
+            original_path = Path(directory).joinpath(path)
+            full_path = original_path.resolve()
+            directory = Path(directory).resolve()
             try:
-                return full_path, os.stat(full_path)
+                stat_result = os.lstat(original_path)
+                full_path.relative_to(directory)
+                return full_path, stat_result
+            except ValueError:
+                # Don't allow misbehaving clients to break out of the static files
+                # directory if not following symlinks.
+                if not stat.S_ISLNK(stat_result.st_mode):
+                    continue
```

Comment:
This conditional is not doing anything, is it? 🤔

If it's true or false, it doesn't matter, it will still continue...

Reply:
Actually this came from a refactor requested to re-use the `lstat` result.
Now that I'm looking at the refactored correct version, I think we're not re-using it, it was better in the first place. Right now it is:

```python
try:
    stat_result = os.lstat(original_path)
     full_path.relative_to(directory)
     return full_path, stat_result
 except ValueError:
    # Allow clients to break out of the static files directory
    # if following symlinks.
    if stat.S_ISLNK(stat_result.st_mode):
         stat_result = os.lstat(full_path)
         return full_path, stat_result
```

So if `lstat` returns a symbolic directory, we have to run another `os.stat` on the real file anyway, I think we're complicating it and it's not worth it. What do you think?

Reply:
I think this version which used the `Path` was better: https://github.com/encode/starlette/pull/1377/commits/3fb9da7e82ce00eb316593eba51f1d2f339c6314

Reply:
IMO, I think the current one is fine. It has less "negative conditionals", which makes the code easier to understand.